Redirects after login

thesuss opened the pull request:
PT Stories:
*  https://www.pivotaltracker.com/story/show/135009145
*  https://www.pivotaltracker.com/story/show/135683887

Changes proposed in this pull request:
1. "after_sign_in_path" for users who are and are not admins. 
2. Minor change of text in landing page
3. information on the 'after login' page for users who are and are not members

thesuss committed:
added a route for the info page, routes after_sign_in path there for users who are not admins, conditionally displays the content of that page

thesuss committed:
simplified the controller method

thesuss committed:
simplified the admin chek in user model

thesuss committed:
added acceptance tests for logging in as admin

weedySeaDragon commented on app/models/user.rb:
```diff
@@ -4,4 +4,8 @@ class User < ApplicationRecord
          :recoverable, :rememberable, :trackable, :validatable
+
+  def admin?
+    self.admin
```
don't need `self`

thesuss committed:
didnt need self

weedySeaDragon commented on app/views/membership_applications/information.html.haml:
```diff
@@ -0,0 +1,14 @@
+
+
+- else
+  %header.entry-header
```
There's too much repeated code.  You need to DRY this up.
The only thing that's different is the text in the`%h1.entry-title`  line, so you should have the conditional affect only that line.  You could use a helper:
`helpers/membership_application_helper.rb`

    def user_or_member_instructions(user)
      user.is_member? ? 'Såhär annonserar du ditt medlemsskap' : 'Såhär ansöker du om medlemsskap'
    end

`%h1.entry-title user_or_member_instructions(current_user)`

We know that there will be different content in the body of this per PT: [Landing page for member with complete  company](https://www.pivotaltracker.com/story/show/135009145), so you'll need to do more conditionals for the body.  With this design, I'd suggest using  partials (perhaps with `content_for:`) and a helper that returns which partial to show based on the role

[If you separate the roles so that you have views/user and views/member, you can use the same layout but populate the header and the body with different content, perhaps using `content_for:`.  It will be cleaner than having the conditionals in this file or in helpers that it calls.   ]

thesuss replied:
I think I went with the third option. 
The hound doesn't seem to mind it.

weedySeaDragon reviewed (changes requested):
Need feature tests to verify that the right page is being viewed based on the current role:  Need to verify that a visitor sees the right page and cannot see the page that a user, member, or admin sees; need to verify that a user sees the right page after logging in and cannot see the page that a member or admin sees; and similar scenarios  for a member and admin.

I'm not clear on your definition of a 'landing' page for each of the authorized (logged in) roles.  For an admin, I don't understand the distinction between their landing page and the membership_applications#index page; likewise what's that distinction between a landing page for users & members and the membership_applications#information page?

If there is a distinction, then what is the landing page for each of the roles above, and when would they _land_ on it?

thesuss committed:
merged with upstream

thesuss committed:
fixed stuff on index after comments form SHF

thesuss committed:
added text for the information page, drye'd it up a bit

thesuss committed:
Merge branch 'develop' into land-admin

weedySeaDragon reviewed (changes requested):
Still must have tests for this!
What is expected when a user logs in?
  ... should see "zzzz" text
 And should not see "yyyyy" text

What is expected when a member logs in?

(I can work with you to write the tests if that's helpful.)

When I do things manually, I do not see the correct page and information.

weedySeaDragon committed:
scenarios to make sure the right page is shown after logging in

thesuss committed:
happy and sad path runs green

weedySeaDragon committed:
add steps to be on and to check the 'member instructions' and 'user instructions' page

weedySeaDragon committed:
also test that we are on the correct page (the user instructions/member instructions)

weedySeaDragon committed:
use the method from User; User should be responsible for knowing how to check, not a view

Let User be responsible for doing whatever is needed to figure out if they are are member. Do not directly check the 'internals' (an attribute) of the User class.  A User may change how this works (exactly what is invovled in determining if a User is a member).  No other class or object should have to know exactly what is invovled -- which attributes are/aren't invovled, etc.  All they should care about is the answer to the question: 'are you a member?'
